mllib/src/main/scala/org/apache/spark/ml/classification/Classifier.scala

```diff
@@ -115,7 +116,16 @@ abstract class ClassificationModel[FeaturesType, M <: ClassificationModel[Featur
         }
         predictUDF(col(getFeaturesCol))
       }
-      outputData = outputData.withColumn(getPredictionCol, predUDF)
+      // determine number of classes either from metadata if provided.
+      val labelSchema = dataset.schema($(labelCol))
+      // extract label metadata from label column if present, or create a nominal attribute
+      // to output the number of labels
+      val labelAttribute = Attribute.fromStructField(labelSchema) match {
```

Comment:
Can this labelAttribute be prepared upon model fitting/construction?  Everything except the name of the output column should be available then.  Btw, the attribute name should be from getPredictionCol, not "label"

Also, can this be computed in validateAndTransformSchema so that transformSchema computes it?